portal-ui/src/screens/Console/Dashboard/DownloadWidgetDataButton.tsx

```diff
@@ -0,0 +1,160 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Fragment } from "react";
+import { Menu, MenuItem, Box } from "@mui/material";
+import ListItemText from "@mui/material/ListItemText";
+import { DownloadIcon } from "../../../icons";
+import { exportComponentAsPNG } from "react-component-export-image";
+import { ErrorResponseHandler } from "../../../common/types";
+import { useAppDispatch } from "../../../../src/store";
+import { setErrorSnackMessage } from "../../../../src/systemSlice";
+interface IDownloadWidgetDataButton {
+  title: any;
+  componentRef: any;
+  data: any;
+}
+
+const DownloadWidgetDataButton = ({
+  title,
+  componentRef,
+  data,
+}: IDownloadWidgetDataButton) => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const openDownloadMenu = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleCloseDownload = () => {
+    setAnchorEl(null);
+  };
+  const download = (filename: string, text: string) => {
+    let element = document.createElement("a");
+    element.setAttribute("href", "data:text/plain;charset=utf-8," + text);
+    element.setAttribute("download", filename);
+
+    element.style.display = "none";
+    document.body.appendChild(element);
+
+    element.click();
+    document.body.removeChild(element);
+  };
+
+  const dispatch = useAppDispatch();
+  const onDownloadError = (err: ErrorResponseHandler) =>
+    dispatch(setErrorSnackMessage(err));
+
+  const convertToCSV = (objectToConvert: any) => {
+    const array = [Object.keys(objectToConvert[0])].concat(objectToConvert);
+    return array
+      .map((it) => {
+        return Object.values(it).toString();
+      })
+      .join(",");
```

Comment:
Each item in the CSV file downloaded needs to be in its own row, currently we have everything in a single row as below:

<img width="956" alt="Screen Shot 2022-08-16 at 4 27 03 PM" src="https://user-images.githubusercontent.com/27314853/185002515-87a5088d-0411-46a4-be9c-7d170c25ea5c.png">

The logic here works great apart from only this `join` statement. We can fix this as follows as each item needs to be in its own row:

```suggestion
      .join("\n");
```

After applying the suggestion, our CSV file would look like the following, where every item is in its own row:

<img width="206" alt="Screen Shot 2022-08-16 at 4 29 55 PM" src="https://user-images.githubusercontent.com/27314853/185002547-22b60ba3-57e5-412f-94d3-175d21097409.png">